### PR TITLE
fix: eval in portableTimingSafeEqual

### PR DESCRIPTION
**Pull request by seebees**
`eval` was added to help ensure that `portableTimingSafeEqual`
is not optimized into a non-constant time function.
However, the Content Security Policy `'unsafe-eval'` will flag this.

It is better to remove this and risk such an optimization,
then to force customers to weaken the Content Security Policy.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: modules/material-management/src/cryptographic_material.ts [hunk 1]
```diff
@@ -82,14 +82,19 @@ const timingSafeEqual: (a: Uint8Array, b: Uint8Array) => boolean = (function ()
...
-      * Side channel attacks are pernicious and subtle.
-      */
-    eval('') // eslint-disable-line no-eval
+     * Adding `eval` could prevent the optimization, but this is no grantee.
+     * The eval below is commented out,
+     * because if a browser is using a Content Security Policy with `'unsafe-eval'`
+     * it would fail on this eval.
+     * The value in attempting to ensure this function is not optimized,
+     * is not worth the cost of making customers to alow `'unsafe-eval'`.
+     * If you copy this function for your own use, make sure to educate yourself.
```
Comment (mattsb42-aws) [inline modules/material-management/src/cryptographic_material.ts]: Do we have a reference for such a person to read up on? Just saying "educate yourself" comes across as dismissive, and I know that was not your intent.

File: modules/material-management/src/cryptographic_material.ts [hunk 2]
```diff
@@ -82,14 +82,19 @@ const timingSafeEqual: (a: Uint8Array, b: Uint8Array) => boolean = (function ()
...
+     * Side channel attacks are pernicious and subtle.
+     */
+    // eval('') // eslint-disable-line no-eval
     /* Check for early return (Postcondition) UNTESTED: Size is well-know information.
```
Comment (mattsb42-aws) [inline modules/material-management/src/cryptographic_material.ts]:
```suggestion
    /* Check for early return (Postcondition) UNTESTED: Size is well-know information
```

File: modules/material-management/src/cryptographic_material.ts (hunk 1 above)
Comment (seebees) [reply modules/material-management/src/cryptographic_material.ts]: This is a good point.  https://codahale.com/a-lesson-in-timing-attacks/ is linked above.  Perhaps adding `please`?  I was hoping that next sentence might mitigate some of this.

Comment (mattsb42-aws) [reply modules/material-management/src/cryptographic_material.ts]: Maybe "please review the above link before use"?

**Pull request merged**